Source/WebKit/WebKitSwift/Preview/PreviewWindowController.swift

```diff
@@ -79,6 +88,28 @@ public final class PreviewWindowController: NSObject {
             }
         }
     }
+    
+    @objc(updateImage:) public func updateImage(url: URL) {
+        self.item = PreviewItem(url: url, displayName: nil, editingMode: .disabled);
+        
+        Task {
+            do {
+                if !self.isOpen {
+                    if let continuation = self.windowOpenedContinuation {
+                        // The Quick Look window isn't ready yet, but there's already been an earlier update queued
+                        // Throw that update and queue this one instead
+                        continuation.resume(throwing: UpdateError.newUpdateQueued)
+                    }
+                    try await withCheckedThrowingContinuation { continuation in
+                        self.windowOpenedContinuation = continuation
+                    }
```

Comment:
maybe we should catch `newUpdateQueued` separately since we use it to cancel the previous update but it shouldn't, I think, be logged as an error

Reply:
Good catch. Changed that case to a debug log